tests/unit/test_charm.py

```diff
@@ -1,18 +1,128 @@
 # Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+from typing import Dict, List, Literal
+from unittest.mock import Mock, patch
+
 import ops
 import ops.testing
 import pytest
+import requests
 from charm import GocertCharm
-from scenario import Context, Event, State
+from scenario import Container, Context, Event, Network, State, Storage
+
+# https://res.cloudinary.com/canonical/image/fetch/f_auto,q_auto/https://discourse-charmhub-io.s3.eu-west-2.amazonaws.com/original/2X/4/4ac42dc8a238a003c7d56fe282246ca102dd594f.png
 
 
 class TestCharm:
-    @pytest.fixture(scope="function", autouse=True)
+    @pytest.fixture(scope="function")
     def context(self):
         yield Context(GocertCharm)
 
-    def test_start_charm(self, context):
-        out = context.run(Event("start"), State())
-        assert out.unit_status == ops.ActiveStatus()
+    @pytest.mark.parametrize(
+        "storages_state,storage_ready",
+        [
+            pytest.param([Storage(name="config")], False, id="config_storage"),
+            pytest.param([Storage(name="database")], False, id="database_storage"),
+            pytest.param(
+                [Storage(name="config"), Storage(name="database")], True, id="both_storages"
+            ),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "containers_state,container_ready",
+        [
+            pytest.param(
+                [Container(name="gocert", can_connect=False)], False, id="container_cant_connect"
+            ),
+            pytest.param(
+                [Container(name="gocert", can_connect=True)], True, id="container_can_connect"
+            ),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "networks_state,network_ready",
+        [
+            pytest.param({"juju-info": Network([], [], [])}, False, id="network_not_available"),
+            pytest.param({"juju-info": Network.default()}, True, id="network_available"),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "gocert_state,gocert_status",
+        [
+            pytest.param(
+                Mock(side_effect=requests.ConnectionError), "not-running", id="gocert_not_running"
+            ),
+            pytest.param(
+                Mock(
+                    **{"json.return_value": {"initialized": False}, "status_code": 200},
+                ),
+                "running",
+                id="gocert_running",
+            ),
+            pytest.param(
+                Mock(
+                    **{"json.return_value": {"initialized": True}, "status_code": 200},
+                ),
+                "initialized",
+                id="gocert_initialized",
+            ),
+        ],
+    )
+    def test_configure_handler(
```

Comment:
Let's not change the convention we have for unit test names please.

Reply:
reverted